[AOTI] Switch to use shim v2

hl475 opened the pull request:
Differential Revision: D56709309




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

desertfire commented on torch/_inductor/config.py:
```diff
@@ -45,7 +45,7 @@ def is_fbcode():
 
 c_shim_version = os.environ.get(
-    "TORCHINDUCTOR_C_SHIM_VERSION", "1" if is_fbcode() else "2"
+    "TORCHINDUCTOR_C_SHIM_VERSION", "1" if (is_fbcode() and torch.version.hip) else "2"
```
We should get rid of `torch.version.hip`, but let's chat offline to figure out how to do it.

hl475 committed:
[AOTI] Switch to use shim v2 (#127674)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/127674

Test Plan:
```
rm -rf /tmp/torchinductor_huaminli/ && TRITON_LOCAL_BUILD=1 buck2 run 'fbcode//mode/opt' fbcode//accelerators/workloads/models/slimdsnn:slimdsnn -- aot
buck2 test -c fbcode.nvcc_arch=a100 'fbcode//mode/opt' fbcode//caffe2/torch/fb/model_transform/experimental/benchmark/test/aotinductor:test_aot_inductor_benchmark -- --exact 'caffe2/torch/fb/model_transform/experimental/benchmark/test/aotinductor:test_aot_inductor_benchmark - test_aot_inductor_benchmark_oemae (caffe2.torch.fb.model_transform.experimental.benchmark.test.aotinductor.test_aot_inductor_benchmark.AOTInductorBenchmark)'
```

Reviewed By: desertfire

Differential Revision: D56709309

hl475 commented:
@pytorchbot merge

hl475 commented:
@pytorchbot label "topic: not user facing"

hl475 commented:
@pytorchbot merge

hl475 commented:
@pytorchbot merge

clee2000 commented:
@pytorchot revert -m "tests failed internally D56709309" -c ghfirst

clee2000 commented:
@pytorchbot revert -m "tests failed internally D56709309" -c ghfirst
